idea/src/org/jetbrains/kotlin/idea/quickfix/RemovePartsFromPropertyFix.java

```diff
@@ -135,4 +136,23 @@ public static KotlinSingleIntentionActionFactory createFactory() {
             }
         };
     }
+
+    public static KotlinSingleIntentionActionFactory createLateInitFactory() {
+        return new KotlinSingleIntentionActionFactory() {
+            @Override
+            public KotlinQuickFixAction<KtProperty> createAction(@NotNull Diagnostic diagnostic) {
+                PsiElement element = Errors.INAPPLICABLE_LATEINIT_MODIFIER.cast(diagnostic).getPsiElement();
+                KtProperty property = PsiTreeUtil.getParentOfType(element, KtProperty.class);
+                if (property == null) return null;
+
+                boolean hasInitializer = property.hasInitializer();
+                boolean hasGetter = property.getGetter() != null && property.getGetter().getBodyExpression() != null;
```

Comment:
I think the getter and setter checks are needed. This quickfix apply to only lateinit property because the element get from INAPPLICABLE_LATEINIT_MODIFIER.